node/src/components/consensus/era_supervisor.rs

```diff
@@ -674,3 +688,43 @@ where
         }
     }
 }
+
+/// Metrics to track rate of finalization
+#[derive(Debug)]
+pub struct EraSupervisorMetrics {
+    /// Gauge to track rate.
+    finalization_time: Gauge,
```

Comment:
`finalization_time` sounds to me as if it measures how long it took to run finality detector but you use `finalized_block.timestamp().elapsed().millis()` for it. The latter will give you the timestamp of the finalized block but it will always be the beginning of the leader round. 

We should report what is measured [here](https://github.com/CasperLabs/casper-node/blob/master/node/src/components/consensus/highway_core/finality_detector.rs#L100-L113).

Reply:
That was initially my idea, given the way metrics are conventially coupled to a component, hooking up a metric to track the FinalityDetector. I shall set it up to track it in finality detector

Reply:
To be honest, "finalization time" sounds right to me: How long it takes the network as a whole to finalize a block.
What you link to I would call something like "finality detector run time".

Reply:
OK, I rest my case then. I agree with Andreas.